snap/validate_test.go

```diff
@@ -225,6 +227,9 @@ version: 1.0
 }
 
 func (s *ValidateSuite) TestIllegalHookName(c *C) {
```

Comment:
We decided earlier that hook names could only consist of [lowercase letters and hyphens](https://github.com/snapcore/snapd/blob/master/snap/validate.go#L30). There are two ways to get hooks into `snap.Info`:

1) Declare in the YAML
2) Ship an executable within `meta/hooks`

There are already tests for making sure hook names from (1) get validated, this is a test making sure names from (2) get validated. Otherwise, if we're careless and whitelist a hook with the pattern ".*foo" it's possible for hook named "123foo" to make it in, even though that hook name is actually invalid.